### PR TITLE
Add `initconfig` command

### LINKED ISSUE #275: Have command to create path /etc/acs

### PR DESCRIPTION
**Pull request by calvin-laurenson**
Honestly I have no idea how to use rust but I was bored so I just googled and tried things until it worked. Supposed to close #275

### CODE REVIEW

**Review by JakeRoggenbuck — COMMENTED** [review]
Hey! Thanks for the contribution. It looks very good. I will test and merge this later tonight, but it looks good. Have a great summer!

File: src/config.rs [hunk 1]
```diff
@@ -36,8 +36,53 @@ pub fn default_config() -> Config {
...
+    if !config_file_exists() {
+        // If the config directory doesn't exist, create it
+        if !config_dir_exists() {
+            let acs_dir = std::fs::create_dir_all("/etc/acs/");
+            match acs_dir {
+                Ok(_) => {}
+                Err(error) => {
+                    match error.kind() {
+                        ErrorKind::PermissionDenied => {
+                            warn_user!("Could not create config directory '/etc/acs/' Permission denied");
```
Comment (Camerooooon) [inline src/config.rs:50]: You should probably tell the user to run this with sudo

File: src/config.rs [hunk 2]
```diff
@@ -36,8 +36,53 @@ pub fn default_config() -> Config {
...
+                    }
+                }
+            }
+        }
+        let config_file = File::create(&config_path());
+        let mut config = match config_file {
+            Ok(file) => file,
+            Err(error) => match error.kind() {
+                ErrorKind::PermissionDenied => {
+                    warn_user!("Looks like you don't have permission to write to /etc/acs/acs.toml");
```
Comment (Camerooooon) [inline src/config.rs:66]: Again notify sudo

File: src/daemon.rs [hunk 3]
```diff
@@ -24,12 +25,16 @@ use super::Error;
...
+#[derive(Clone, Debug, PartialEq, Serialize)]
 pub enum State {
     Normal,
+    #[serde(rename = "battery_percent_rule")]
     BatteryLow,
+    #[serde(rename = "lid_open_rule")]
     LidClosed,
+    #[serde(rename = "ac_charging_rule")]
     Charging,
+    #[serde(rename = "cpu_usage_rule")]
```
Comment (Camerooooon) [inline src/daemon.rs:37]: What does this do?

**Review by Camerooooon — CHANGES_REQUESTED** [review]
Thank you so much for doing this Calvin! This looks amazing, I have a few minor user message changes that I think you should add. Additionally, in src/config.rs on line 198, you should change the warning message for when a user attempts to run acs without a config to something that hints at the existence of the initconfig command.

Thank you so so much for contributing to acs this is some amazing code, especially for someone who is just beginning rust!

File: src/daemon.rs (hunk 3 above)
Comment (Camerooooon) [reply src/daemon.rs:37]: oh wait I see nevermind!

**Comment by Camerooooon** [comment]
The CI failed because you need to run `cargo fmt` in order to format the code as per the rust style guides. Tests passed on my machine so that's the only problem as of now.

**Review by JakeRoggenbuck — APPROVED** [review]
Great, Thanks!

**Review by Camerooooon — APPROVED** [review]
Wonderful!

**Pull request merged**